webapp/login.go

```diff
@@ -0,0 +1,255 @@
+package webapp
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/gob"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-github/github"
+	"github.com/gorilla/securecookie"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"golang.org/x/oauth2"
+	ghOAuth "golang.org/x/oauth2/github"
+	"google.golang.org/appengine"
+)
+
+// Copyright 2019 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+func init() {
+	gob.Register(map[string]interface{}{})
+	// Need RegisterName - for local packages, Register appends main[0-9]{5}.User
+	gob.RegisterName("User", User{})
+}
+
+// User represents an authenticated GitHub user.
+type User struct {
+	GitHubHandle string
+	GithuhEmail  string
+}
+
+func loginHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := shared.NewAppEngineContext(r)
+	aeAPI := shared.NewAppEngineAPI(ctx)
+	if !aeAPI.IsFeatureEnabled("githubLogin") {
+		http.Error(w, "Feature not implemented", http.StatusNotImplemented)
+		return
+	}
+
+	user, token := getUserFromCookie(r)
+	returnURL := r.FormValue("return")
+	redirect := returnURL
```

Comment:
Nit: I'd prefer setting `redirect` to an empty string here (effectively only declaring `redirect`), as it'd be set to completely different values in the two branches below.